### ISSUE TITLE
Beacon SDK initialization exceptions from robolectric 

### ISSUE THREAD

**Issue by theBradfo** [helpscout/beacon-android-sdk-sample#79]
* Beacon version: 2.0.1
* Robolectric version: 4.1

**Steps to reproduce**
1. Initialize the beacon sdk in your application `onCreate` like:
```
            new Beacon.Builder()
                    .withContext(this)
                    .withBeaconId(<any id>)
                    .build();
```

2. Observe the following exception when the robolectric test runner attempts to initialize the beacon sdk:
```
com.helpscout.beacon.SDKInitException: Beacon not initialized. If you have disabled auto initialization via manifest meta data, ensure Beacon is manually init using Beacon.Builder().withContext(context)
	at com.helpscout.beacon.Beacon.datastore(Beacon.java:55)
	at com.helpscout.beacon.Beacon.initDatastoreValues(Beacon.java:406)
	at com.helpscout.beacon.Beacon.<init>(Beacon.java:44)
	at com.helpscout.beacon.Beacon.<init>(Beacon.java:29)
	at com.helpscout.beacon.Beacon$Builder.build(Beacon.java:497)
	at com.adhoclabs.burner.BurnerApplication.onCreate(BurnerApplication.java:185)
	at org.robolectric.android.internal.ParallelUniverse.lambda$setUpApplicationState$0(ParallelUniverse.java:246)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:72)
	at org.robolectric.android.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:246)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:379)
```

Note that this same exception occurs _with or without_ the application metadata tag:
```
        <meta-data android:name="com.helpscout.beacon.SDKAutoInitialise" android:value="false" />
```

Debugging this at runtime, it seems the `BeaconInitProvider.initDataStore(...)` initializes and returns a `BeaconPreferences` reference, but the caller `build()` does nothing with this reference.  The `Beacon.datastore` reference is thus `null` and this code flow fails when it attempts to initialize the `Beacon` instance with the provided `Beacon.Builder`.

**Comment by scottyab** [comment]
Hey @theBradfo thanks for logging this. In 2.0.1 we moved the way we expose the BeaconDatastore (BeaconPreferences) to try and reduce its visibility as it wasn't mean to be readable/editable but users of the SDK. Sorry our move has triggered a Robolectric crash for you. We'll take a look. In the meantime if you aren't using our realtime Chat yet, then recommend dropping back to 2.0.0 for the time being.

**Comment by theBradfo** [comment]
Thanks for the quick response @scottyab, however after dropping to `2.0.0` i'm still receiving the same exception at robolectric test runtime.  I can confirm we are not presently using the real time chat features.

**Comment by jotaemepereira** [comment]
The issue on our internal tracking system: https://helpscout.atlassian.net/browse/ANDSDK-282

**Comment by jotaemepereira** [comment]
@theBradfo Hey! Sorry for the delay in the response. As you noticed, we are not setting the `datastore` when creating a `Beacon` with the `withContext` method. We are currently reviewing a fix that if it gets approved, it should go in the next release. 

In the meantime, you can use the following workaround in your Roboelectric tests: Roboelectric provides you with a `@Config` annotation where you can choose an application to run your tests.
There you can set a fake Android application where you will not have any `Beacon` dependency; thus, you will not experience the exception.

Let me know if that works, and thanks for raising the issue!

**Comment by aliripe** [comment]
Hi @theBradfo thanks again for raising this issue. This has been fixed as part of the 2.0.3 release.

Let us know if it works for you now!
